components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/KeyStoreBasedExternalCryptoProvider.java

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.crypto.provider;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.core.util.KeyStoreManager;
+import org.wso2.carbon.crypto.api.CertificateInfo;
+import org.wso2.carbon.crypto.api.CryptoContext;
+import org.wso2.carbon.crypto.api.CryptoException;
+import org.wso2.carbon.crypto.api.ExternalCryptoProvider;
+import org.wso2.carbon.crypto.api.PrivateKeyInfo;
+
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.cert.Certificate;
+import javax.crypto.Cipher;
+
+import static org.wso2.carbon.base.MultitenantConstants.SUPER_TENANT_ID;
+
+/**
+ * The Java Keystore based implementation of {@link ExternalCryptoProvider}
+ */
+public class KeyStoreBasedExternalCryptoProvider implements ExternalCryptoProvider {
+
+    private static Log log = LogFactory.getLog(KeyStoreBasedExternalCryptoProvider.class);
+
+    public KeyStoreBasedExternalCryptoProvider() {
```

Comment:
Do we need to define the no argument constructor here since we do not have any other constructors defined for this class?